vcd/datasource_vcd_catalog_media_test.go

```diff
@@ -0,0 +1,147 @@
+// +build catalog ALL functional
+
+package vcd
+
+import (
+	"fmt"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+// Test catalog and catalog media data sources
+// Using a catalog data source we reference a catalog media data source
+// Using a catalog media data source we create another catalog media
+// where the description is the first data source ID
+func TestAccVcdCatalogAndMediaDatasource(t *testing.T) {
+	var TestCatalogMediaDS = "TestCatalogMediaDS"
+	var TestAccVcdDataSourceMedia = "TestAccVcdCatalogMediaBasic"
+	var TestAccVcdDataSourceMediaDescription = "TestAccVcdCatalogMediaBasicDescription"
+
+	var media govcd.Media
+
+	var params = StringMap{
+		"Org":              testConfig.VCD.Org,
+		"Catalog":          testSuiteCatalogName,
+		"NewCatalogMedia":  TestCatalogMediaDS,
+		"OvaPath":          testConfig.Ova.OvaPath,
+		"UploadPieceSize":  testConfig.Ova.UploadPieceSize,
+		"UploadProgress":   testConfig.Ova.UploadProgress,
+		"Tags":             "catalog",
+		"CatalogMediaName": TestAccVcdDataSourceMedia,
+		"Description":      TestAccVcdDataSourceMediaDescription,
+		"MediaPath":        testConfig.Media.MediaPath,
+	}
+
+	configText := templateFill(testAccCheckVcdCatalogMediaDS, params)
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { preRunChecks(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: catalogMediaDestroyed(testSuiteCatalogName, TestCatalogMediaDS),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config:             configText,
+				ExpectNonEmptyPlan: true,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVcdCatalogMediaExists("vcd_catalog_media."+TestAccVcdDataSourceMedia, &media),
+					resource.TestMatchOutput("owner_name", regexp.MustCompile(`^\w*$`)),
```

Comment:
This regular expression means "match a string of  **zero** or more word characters". Thus it will match "abc", or "a_bc", but also the empty string "". However, it won't match "ab-c"

Reply:
Do we now user input restrictions? or just check non empty string?

Reply:
updated to `non-empty`